server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/helper/ArgumentHelper.java

```diff
@@ -304,6 +304,9 @@ private Object correctObjectClass(Object argumentValue, Field field, DataFetchin
         } else if (receivedClassName.equals(String.class.getName())) {
             // We got a String, but not expecting one. Lets bind to Pojo with JsonB
             // This happens with @DefaultValue and Transformable (Passthrough) Scalars
+            if (field.getReference().getClassName().equals("org.bson.types.ObjectId")) { // edge case for ObjectId
```

Comment:
Could you add a little  bit more of an explanation into the comment?
Something along the lines that we need to avoid letting JSON-B call the no-arg constructor of ObjectId, because then a new, random ID would get generated.